operator/v1/types_console.go

```diff
@@ -375,6 +380,34 @@ const (
 	BrandROSA Brand = "ROSA"
 )
 
+// Ingress allows cluster admin to configure alternative ingress for the console.
+type Ingress struct {
+	// consoleURL is a URL to be used as the base console address.
+	// If not specified, the console route hostname will be used.
+	// This field is required for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Make sure that appropriate ingress is set up at this URL before assigning the value,
+	// otherwise the console operator may go degraded.
+	// Must use the HTTPS scheme.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="isURL(self)",message="console url must be a valid absolute URL"
+	// +kubebuilder:validation:XValidation:rule="url(self).getScheme() == 'https'",message="console url scheme must be https"
+	// +kubebuilder:validation:MaxLength=1024
+	ConsoleURL string `json:"consoleURL,omitempty"`
```

Comment:
Since these two URLs (and the object itself) are part of a config API, we tend to prefer to get rid of the omitempty.

If you were to do that, you would have to prefix the CEL to avoid running the validation when the string has length 0.

Can you give this a go?

The benefit here is that there's improved visibility and end users will see the fields in the rendered config as empty values, so they can see that the option exists and maybe then want to populate it

Reply:
> The benefit here is that there's improved visibility and end users will see the fields in the rendered config as empty values, so they can see that the option exists and maybe then want to populate it

What's wrong with the good old `oc explain`?

Reply:
Sometimes a user may have to go through several levels of oc explain to understand what options are available. Printing the whole struct out in configuration APIs makes it clear that there are levels below what they might see that they had no idea they might even want to `oc explain`

Reply:
Doesn't this look a little misleading? When the CEL validation wants the value to be an URL but the user sees an empty string because they omitted the field.

Reply:
I would argue that users seeing an empty URL as allowed would indicate that this is an empty value that means nothing is going to happen no? It's only required to be a URL when it's length is non-zero I think is a reasonable validation no?

Reply:
Done, including `spec.ingress`.